src/pyodide/python.js

```diff
@@ -0,0 +1,107 @@
+import { _createPyodideModule } from "pyodide-internal:pyodide-bundle/pyodide.asm";
+import module from "pyodide-internal:pyodide-bundle/pyodide.asm.wasm";
+import stdlib from "pyodide-internal:pyodide-bundle/python_stdlib.zip";
+
+let memory = undefined;
+
+function prepareFileSystem(Module) {
+  const pymajor = 3;
+  const pyminor = 11;
+  Module.FS.mkdirTree("/lib");
+  Module.FS.mkdirTree(`/lib/python${pymajor}.${pyminor}/site-packages`);
+  Module.FS.writeFile(
+    `/lib/python${pymajor}${pyminor}.zip`,
+    new Uint8Array(stdlib),
+    { canOwn: true }
+  );
+  Module.FS.mkdir(Module.API.config.env.HOME);
+}
+
+async function makeSnapshot(Module, run) {
+  const imports = [
+    "_pyodide.docstring",
+    "_pyodide._core_docs",
+    "traceback",
+    "collections.abc",
+    "asyncio",
+    "inspect",
+    "tarfile",
+    "importlib.metadata",
+    "re",
+    "shutil",
+    "sysconfig",
+    "importlib.machinery",
+    "pathlib",
+    "site",
+    "tempfile",
+    "typing",
+    "zipfile",
+  ];
+  const to_import = imports.join(",");
+  const to_delete = Array.from(
+    new Set(imports.map((x) => x.split(".")[0]))
+  ).join(",");
+  run(`import ${to_import}`);
+  run("sysconfig.get_config_vars()");
+  run(`del ${to_delete}`);
+  memory = Module.HEAP8.slice();
+}
+
+export async function loadPyodide() {
+  const API = {};
+  const config = { jsglobals: globalThis, env: { HOME: "/session" } };
+  API.config = config;
+  const Module = {
+    noInitialRun: !!memory,
+    API,
+    locateFile: (x) => x,
+    instantiateWasm(info, receiveInstance) {
+      (async function () {
+        const instance = await WebAssembly.instantiate(module, info);
```

Comment:
This is using the alternative Proxies `WebAssembly` correct? I'd generally prefer that we use a different name for it.